addons/account/data/invoice_action_data.xml

```diff
@@ -52,18 +52,7 @@
     </div>
     <div style="width: 347px; margin: 0px; padding: 5px 14px; line-height: 16px; background-color: #F2F2F2;">
         <span style="color: #222; margin-bottom: 5px; display: block; ">
-        % if object.company_id.street:
-            ${object.company_id.street}<br/>
-        % endif
-        % if object.company_id.street2:
-            ${object.company_id.street2}<br/>
-        % endif
-        % if object.company_id.city or object.company_id.zip:
-            ${object.company_id.zip} ${object.company_id.city}<br/>
-        % endif
-        % if object.company_id.country_id:
-            ${object.company_id.state_id and ('%s, ' % object.company_id.state_id.name) or ''} ${object.company_id.country_id.name or ''}<br/>
-        % endif
+            ${object.company_id.partner_id.sudo().with_context(show_address=True).name_get()[0][1].replace('\n', '<br/>') | safe}
```

Comment:
Couldn't you add a html key in the context the does the formatting directly in the python code, instead of doing replace in the report ?